imap_processing/lo/l1a/lo_l1a.py

```diff
@@ -0,0 +1,174 @@
+    ----------
+    dependency :
+        dependency file needed for data product creation.
+        Should always be only one for L1A.
+
+    Returns
+    -------
+    created_file_paths: list[Path]
+        location of created CDF files
+    """
+    # TODO: decom the CCSDS file
+
+    # TODO: sort the decommed packets by SHCOARSE
+
+    # TODO: group the packets by APID
+
+    # TODO: Loop through packets
+    # TODO: Inside Loop: Check for each APID and Initialize data
+    #  class object with packet contents, add to list
+
+    # create the attribute manager for this data level
+    attr_mgr = ImapCdfAttributes()
+    attr_mgr.add_instrument_global_attrs(instrument="lo")
+    attr_mgr.add_instrument_variable_attrs(instrument="lo", level="l1a")
+
+    # TODO: replace with real processing when sample data is available
+    # Temporary code until I get sample data. The fields will still be pulled
+    # from the data classes, but the data class will be determined by the
+    # APID in the decommed file and the data class will first be populated
+    # with the packet data.
+    packet_file_name = dependency.name
+    if "_de_" in packet_file_name:
+        data_fields = fields(ScienceDirectEvents)
+        logical_source = "imap_lo_l1a_de"
+    elif "_spin_" in packet_file_name:
+        # Spin data class doesn't exist yet. fake the fields() output
+        data_field_tup = namedtuple("data_field_tup", ["name"])
+        data_fields = [
+            data_field_tup("NUM_COMPLETED"),
+            data_field_tup("ACQ_END"),
+            data_field_tup("SPIN_SECONDS"),
+            data_field_tup("SPIN_SUBSECONDS"),
+            data_field_tup("ESA_P_DAC"),
+            data_field_tup("ESA_N_DAC"),
+            data_field_tup("VAL_PERIOD"),
+            data_field_tup("VAL_SPIN"),
+            data_field_tup("SOURCE"),
+        ]
+        logical_source = "imap_lo_l1a_spin"
+
+    dataset = create_datasets(attr_mgr, logical_source, data_fields)
+    create_file_paths = write_cdf(dataset)
+    return create_file_paths
+
+
+# TODO: This is going to work differently when I sample data.
+#  The data_fields input is temporary.
+def create_datasets(attr_mgr, logical_source, data_fields):
+    """Create a dataset using the populated data classes.
+
+    Parameters
+    ----------
+    attr_mgr : ImapCdfAttributes
+        attribute manager used to get the data product field's attributes
+    logical_source: str
+        The logical source of the data product that's being created
+    data_fields: list[dataclasses.Field]
+
+    Returns
+    -------
+    xr.dataset
+        dataset with all data product fields in xr.DataArray
+    """
+    # Convert each packet's spacecraft time to an absolute epoch time
+    # TODO: replace temp hardcoded values with packet values
+    epoch_converted_time = [calc_start_time(time) for time in [0, 1, 2]]
+
+    # Create a data array for the poch time
+    # TODO: might need to update the attrs to use new YAML file
```

Comment:
Yes, this should I think, especially if you've already got the other ones using the new style YAML definitions.

But I don't see an `epoch` variable in your variables file, so you might need to add it to the YAML definition still.

Reply:
Yeah, I left that out because I thought we'd be adding a common one somewhere. But I just added a epoch to the YAML for now